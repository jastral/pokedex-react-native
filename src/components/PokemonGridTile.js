import {
  Pressable,
  View,
  Text,
  StyleSheet,
  Platform,
  Image,
} from "react-native";

import { LinearGradient } from "expo-linear-gradient";

import { colorTypeGradients } from "../utils/utils";

export default function PokemonGridTile({ name, pokeId, onPress, type }) {
  let finalGradient;

  if (type.length === 2) {
    finalGradient = colorTypeGradients(
      type[0].type.name,
      type[1].type.name,
      type.length
    );
  } else {
    finalGradient = colorTypeGradients(
      type[0].type.name,
      type[0].type.name,
      type.length
    );
  }

  return (
    <LinearGradient
      style={styles.gridItem}
      colors={[finalGradient[0], finalGradient[1]]}
    >
      <Pressable
        android_ripple={{ color: "#ccc" }}
        style={({ pressed }) => [
          styles.button,
          pressed ? styles.buttonPressed : null,
        ]}
        onPress={onPress}
      >
        <View style={styles.innerContainer}>
          <Image
            style={styles.tinyLogo}
            source={{
              uri: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokeId}.png`,
            }}
          />
          <Text style={styles.name}>{name.toUpperCase()}</Text>
          <Text style={styles.name}>{pokeId}</Text>
        </View>
      </Pressable>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  gridItem: {
    flex: 1,
    margin: 16,
    height: 250,
    borderRadius: 8,
    backgroundColor: "white",
    elevation: 4,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.25,
    shadowRadius: 8,
    overflow: Platform.OS === "android" ? "hidden" : "visible",
    padding: 8,
    justifyContent: "center",
  },
  button: {
    flex: 1,
  },
  buttonPressed: {
    opacity: 0.5,
  },
  innerContainer: {
    flex: 1,
    padding: 16,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 8,
    overflow: "hidden",
  },
  name: {
    fontSize: 18,
    fontWeight: "bold",
  },
  tinyLogo: {
    width: 150,
    height: 150,
  },
});
