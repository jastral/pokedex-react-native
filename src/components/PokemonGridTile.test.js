import React from "react";
import { ActivityIndicator, FlatList, Text, TextInput } from "react-native";
import renderer from "react-test-renderer";
import PokemonGridTile from "./PokemonGridTile";

const pressHandler = jest.fn();

it("renders correctly", () => {
  const tree = renderer.create(
    <PokemonGridTile
      name="charmander"
      pokeId={4}
      onPress={() => {}}
      type={[{ type: { name: "fire" } }]}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


it("renders correctly", () => {
  const tree = renderer.create(
    <PokemonGridTile
      name="bulbasaur"
      pokeId={1}
      onPress={() => {}}
      type={[{ type: { name: "grass" } }, { type: { name: "poison" } }]}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});