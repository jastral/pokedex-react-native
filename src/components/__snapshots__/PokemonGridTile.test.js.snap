// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4294949233,
        4294949233,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 8,
      "elevation": 4,
      "flex": 1,
      "height": 250,
      "justifyContent": "center",
      "margin": 16,
      "overflow": "visible",
      "padding": 8,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 8,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        null,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 8,
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
          "padding": 16,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png",
          }
        }
        style={
          {
            "height": 150,
            "width": 150,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        CHARMANDER
      </Text>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        4
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`renders correctly 2`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4289265560,
        4292256484,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 8,
      "elevation": 4,
      "flex": 1,
      "height": 250,
      "justifyContent": "center",
      "margin": 16,
      "overflow": "visible",
      "padding": 8,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 8,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        null,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 8,
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
          "padding": 16,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
          }
        }
        style={
          {
            "height": 150,
            "width": 150,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        BULBASAUR
      </Text>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        1
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
